### PR TITLE
[Pthread] Updated operational model for spurious wakeup to cond_wait

### PR DESCRIPTION
**Pull request by lucasccordeiro**

### CODE REVIEW

**Comment by lucasccordeiro** [comment]
This regression test is incorrectly labelled: https://github.com/esbmc/esbmc/blob/master/regression/esbmc-unix/03_atom001a/main.c

Even the comments inside the C file says it:

```
// This pattern comes from a real bug in the Mozilla Application
// Suite. When thread 2 violates the atomicity of thread 1's accesses
// to getCurrentScript, the program crashes.
```

**Review by rafaelsamenezes — APPROVED** [review]
LGTM

File: src/c2goto/library/pthread_lib.c [hunk 1]
```diff
@@ -578,6 +578,22 @@ __ESBMC_HIDE:;
...
+  * is true if the thread should proceed.
+  *
+  * Spurious wakeups from the pthread_cond_wait() or pthread_cond_timedwait()
+  * functions may occur. Since the return from pthread_cond_wait() or pthread_cond_timedwait()
+  * does not imply anything about the value of this predicate, the predicate should be re-evaluated
+  * upon such return.
+  *
+  */
+  _Bool spourious_wakeup = nondet_bool();
+  signalled |= spourious_wakeup;
```
Comment (fbrausse) [inline src/c2goto/library/pthread_lib.c]: typo: spurious.

**Review by fbrausse — COMMENTED** [review]
You changed flags for a lot of the regression tests, what is the reason?

**Comment by lucasccordeiro** [comment]
> You changed flags for a lot of the regression tests, what is the reason?

I have disabled the unwinding assertion checks. Since we can now have spurious wakeups from the `pthread_cond_wait()`, which are inside loops, we might also have execution paths that violate those assertions.

**Comment by fbrausse** [comment]
I see, many thanks for the explanation!

**Review by fbrausse — APPROVED** [review]

File: src/c2goto/library/pthread_lib.c (hunk 1 above)
Comment (lucasccordeiro) [reply src/c2goto/library/pthread_lib.c]: Fixed.

**Pull request merged**